### PR TITLE
feat: diagnostic markers + RPC contract tests (#650, #672)

### LINKED ISSUE #650: Data Explorer: Add Monaco diagnostic markers (squiggles) for syntax errors
## Summary

Add live syntax validation to the Data Explorer's Monaco editor that underlines errors with squiggly lines before query execution, using `monaco.editor.setModelMarkers()`.

## Context

Currently, syntax errors are only shown after query execution in a separate error div. Monaco supports inline diagnostic markers that could show errors as the user types.

## Requirements

- Call `setModelMarkers()` with error position info (line/column) when available
- Also show line/column position in the error display after execution (DE-44 from UX polish audit)
- Requires investigation: does the daemon return parse errors with line/column positions, or would a new validation-only endpoint be needed?

## Source

UX Polish Audit finding DE-44: `docs/plans/2026-03-18-ux-polish-findings.md`

## Priority

Nice-to-have — not blocking release.

### LINKED ISSUE #672: test: Add RPC/daemon contract tests for Extension ↔ CLI protocol
## Description

Create a contract test suite that validates the RPC protocol between the VS Code Extension and the CLI daemon process. This prevents protocol drift between the two surfaces.

## Problem

The Extension communicates with the CLI daemon via StreamJsonRpc. Currently only 3 daemon tests exist (DaemonIntegrationTests, ProtocolContractTests, DaemonTestFixture). If the daemon adds/removes/changes RPC methods or response shapes, the Extension could break silently — the failure only surfaces at runtime.

## Solution

Contract tests that validate:
1. **Method inventory** — every RPC method the Extension calls exists on the daemon
2. **Request schemas** — request parameters match expected types
3. **Response schemas** — response shapes match what the Extension deserializes
4. **Error contracts** — error codes and shapes are consistent

### Approach

- Use the existing DaemonTestFixture to spawn a real daemon
- Call each RPC method with known inputs
- Validate response structure (not just that it doesn't throw)
- Optionally: generate a contract manifest that both sides can validate against

## Acceptance Criteria

- [ ] Every RPC method the Extension calls has a corresponding contract test
- [ ] Tests validate request/response shapes, not just invocation success
- [ ] Tests run without a Dataverse connection (mock/offline mode)
- [ ] All tests pass with `dotnet test --filter "Category=Unit"`

## Notes

- Complements existing DaemonIntegrationTests (which test protocol basics)
- The Extension side could have corresponding TypeScript contract tests (Vitest)
- Consider generating the contract from the daemon's RPC interface via reflection

### PR DESCRIPTION
**Pull request by joshsmithxrm**
## Summary

- Add live SQL diagnostic markers (squiggles) in the Data Explorer Monaco editor via a new `query/validate` RPC endpoint with 300ms debounced validation, request staleness checks, and 3s response timeout
- Enrich execution-time `queryError` responses with structured diagnostics from parse failures
- Add RPC contract test infrastructure: reflection-based method inventory test covering all 97 RPC methods + 8 new Tier 1 shape tests for auth, env, and query endpoints

Closes #650
Closes #672

## Test Plan

- [x] 8 new QueryValidateHandler unit tests (valid SQL, invalid SQL, FetchXML, edge cases)
- [x] 1 new AllRpcMethods_MatchCheckedInInventory reflection test
- [x] 8 new ProtocolContractTests shape tests (auth/select, env/list, env/who, env/select, query/validate, query/validate invalid, query/complete, query/sql)
- [x] Enhanced RC-09 ValidationError_HasExpectedFormat with ErrorData verification
- [x] All 450 extension unit tests pass
- [x] All .NET unit tests pass across net8.0/net9.0/net10.0

## Verification

- [x] /gates passed (all 10 gates green)
- [x] /verify completed (surfaces: extension)
  - Invalid SQL → markers appear (confirmed via `getModelMarkers()`)
  - Fixed SQL → markers clear (empty array)
  - Short SQL (< 3 chars) → no validation triggered
  - FetchXML → no false squiggles
  - Clean daemon logs, no runtime errors
- [x] Pre-PR self-review (0 defects, 2 concerns — F-1 fixed, F-2 accepted as heuristic)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/PPDS.Extension/src/panels/webview/query-panel.ts [hunk 1]
```diff
@@ -321,6 +327,27 @@ if (editor) editor.onDidChangeModelContent(() => {
...
+            validationRequestId++;
+            vscode.postMessage({
+                command: 'requestValidation',
+                requestId: validationRequestId,
+                sql,
+                language,
+            } as QueryPanelWebviewToHost);
+            validationResponseTimeout = setTimeout(() => {
+                if (model) monaco.editor.setModelMarkers(model, 'ppds', []);
+            }, 3000);
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply src/PPDS.Extension/src/panels/webview/query-panel.ts]: Fixed in 9fd967e0f220e67be28ae4fe40e39ac1f3641539 — Removed redundant validationResponseTimeout. Markers are cleared on every keystroke (line 332) before the new validation cycle begins, and the requestId guard in validationResult prevents stale responses from applying markers. The 3s response timeout could only fire when no response arrived â€” at which point markers were already empty from the originating keystroke. No path leaves stale markers visible.
